lockkeys@vaina.lt/extension.js

```diff
@@ -294,6 +297,25 @@ class VisibilityIndicator extends GObject.Object{
 	}
 });
 
+const VisibilityIndicatorCapslock = GObject.registerClass(
+class VisibilityIndicatorCapslock extends GObject.Object{
+	_init(panelButton) {
+		this.panelButton = panelButton;
+		this.config = panelButton.config;
+		this.capsIcon = panelButton.capsIcon;
+	}
+
+	displayState(numlock_state, capslock_state) {
+		if (capslock_state) {
+			this.capsIcon.set_gicon(this.panelButton.getCustIcon('capslock-enabled-symbolic'));
```

Comment:
interesting, why the icon needs to be set each time...

Reply:
Could you try to move it to the constructor and check if that works?

Reply:
I was wondering about that too. Figured there must be some reasoning behind it.

Reply:
Moved the icon setting to the constructors in both VisibilityIndicator classes. It does make more sense that way.